acceleration/fast_inference_tutorial/fast_inference_tutorial.ipynb

```diff
@@ -0,0 +1,646 @@
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_inference(data_list, infer_transforms, model, device, benchmark_type):\n",
+    "    total_time_dict = {}\n",
+    "    roi_size = (96, 96, 96)\n",
+    "    sw_batch_size = 4\n",
+    "\n",
+    "    for idx, sample in enumerate(data_list[:10]):\n",
+    "        start = timer()\n",
+    "        data = infer_transforms({\"image\": sample})\n",
+    "\n",
+    "        with torch.no_grad():\n",
+    "            input_image = (\n",
+    "                data[\"image\"].unsqueeze(0).to(device)\n",
+    "                if benchmark_type in [\"trt\", \"original\"]\n",
+    "                else data[\"image\"].unsqueeze(0)\n",
+    "            )\n",
+    "\n",
+    "            output_image = sliding_window_inference(input_image, roi_size, sw_batch_size, model)\n",
+    "            output_image = output_image.cpu()\n",
+    "\n",
+    "            end = timer()\n",
+    "\n",
+    "        print(output_image.mean())\n",
+    "\n",
+    "        del data\n",
+    "        del input_image\n",
+    "        del output_image\n",
+    "        torch.cuda.empty_cache()\n",
+    "        gc.collect()\n",
+    "\n",
+    "        sample_name = sample.split(\"/\")[-1]\n",
+    "        if idx > 0:\n",
+    "            total_time_dict[sample_name] = end - start\n",
+    "            print(end - start)\n",
+    "    return total_time_dict"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Running the Benchmark\n",
+    "\n",
+    "The cell below will execute the benchmark based on the `benchmark_type` variable.\n",
+    "\n",
+    "#### (Optional) Using the Python Script\n",
+    "\n",
+    "For convenience, a Python script, [`run_benchmark.py`](./run_benchmark.py), is available to run the benchmark. You can open a terminal and execute the following command to run the benchmark for all benchmark types:\n",
+    "\n",
+    "\n",
+    "```bash\n",
+    "for benchmark_type in \"original\" \"trt\" \"trt_gpu_transforms\" \"trt_gds_gpu_transforms\"; do\n",
+    "    python run_benchmark.py --benchmark_type \"$benchmark_type\"\n",
+    "done\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gpu_transforms_flag = False\n",
+    "gpu_loading_flag = False\n",
+    "trt_flag = False\n",
+    "\n",
+    "if \"trt\" in benchmark_type:\n",
+    "    trt_flag = True\n",
+    "if \"gpu_transforms\" in benchmark_type:\n",
+    "    gpu_transforms_flag = True\n",
+    "if \"gds\" in benchmark_type:\n",
+    "    gpu_loading_flag = True\n",
+    "\n",
+    "infer_transforms = get_transforms(device, gpu_loading_flag, gpu_transforms_flag)\n",
+    "model = get_model(device, weights_path, trt_model_path, trt_flag)\n",
+    "\n",
+    "# total_time_dict = run_inference(train_files, infer_transforms, model, device, benchmark_type)\n",
```

Comment:
Why comment these lines?

Thanks.

Reply:
Hi @Nic-Ma , to avoid restarting the notebook multiple times, it's recommended to use cli to run py script for getting benchmark results.